src/lib/udev/google_nvme_id

```diff
@@ -0,0 +1,305 @@
+  return 0
+}
+
+#######################################
+# Ensures a given command/path exists
+# Arguments:
+#   The path for the command
+# Returns:
+#   0 if the command is found/executable and 1 on error
+#######################################
+function confirm_command() {
+  local command_to_check="${1}"
+  command -v "${command_to_check}" > /dev/null 2>&1
+  if [[ $? != 0 ]]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+function print_help_message() {
+  echo "Usage: google_nvme_id [-s] [-h] -d device_path"
+  echo "  -d <device_path> (Required): Specifies the path to generate a name"
+  echo "        for.  This needs to be a path to an nvme device or namespace"
+  echo "  -s: Create symbolic link for the disk under /dev/disk/by-id."
+  echo "        Otherwise, the disk name will be printed to STDOUT"
+  echo "  -h: Print this help message"
+}
+
+function main() {
+  local opt_gen_symlink='false'
+  local device_path=''
+
+  while getopts :d:sh flag; do
+    case "${flag}" in
+      d) device_path="${OPTARG}";;
+      s) opt_gen_symlink='true';;
+      h) print_help_message
+         return 0
+         ;;
+      :) echo "Invalid option: ${OPTARG} requires an argument" 1>&2
+         return 1
+         ;;
+      *) return 1
+    esac
+  done
+
+  if [[ -z "$device_path" ]]; then
+    echo "Device path (-d) argument required. Use -h for full usage." 1>&2
+    exit 1
+  fi
+
+  confirm_command "${nvme_cli_bin}"
+  if [[ $? != 0 ]]; then
+    local err_msg="The nvme utility (/usr/sbin/nvme) was not found."
+    err_msg="${err_msg} You may need to run with sudo or install nvme-cli."
+    err "${err_msg}"
+    return 1
+  fi
+
+  # Ensure the passed device is actually an NVMe device
+  validate_nvme_device "${device_path}"
+  if [[ $? != 0 ]]; then
+    local err_msg="Passed device was not an NVMe device.  (You may need to run"
+    err_msg="${err_msg} this script as root/with sudo)."
+    err "${err_msg}"
+    return 1
+  fi
+
+  # Detect the type of attached nvme device
+  local device_type="$(detect_pd_or_local_ssd "${device_path}")"
+  local device_type_ret_value=$?
+  if [[ "${device_type}" == "${LOCAL_SSD}" ]]; then
+    err "This script only supports PD NVMe disks but given path was Local SSD"
+    return 1
+  elif [[ "${device_type_ret_value}" != 0 ]]; then
+    err "Unknown device type (neither a PD nor Local SSD NVMe device)"
+    return 1
+  fi
+
+  # Fill the global variables for the id command for the given disk type
+  identify_pd_disk "${device_path}"
+  identify_ret_value=$?
+
+  # Error messages will be printed closer to error, no need to reprint here
+  if [[ "${identify_ret_value}" != 0 ]]; then
+    return "${identify_ret_value}"
+  fi
+
+  # Gen symlinks or print out the globals set by the identify command
+  if [[ "${opt_gen_symlink}" == 'true' ]]; then
+    gen_symlink "${device_path}"
+  else
+    echo "ID_SERIAL_SHORT=${ID_SERIAL_SHORT}"
+    echo "ID_SERIAL=${ID_SERIAL}"
```

Comment:
these are the primary outputs of the script in normal operation, right?

Reply:
Yep.  They get consumed by the updated udev rules.